lib/services/analytics/analytics-service.ts

```diff
@@ -1,46 +1,73 @@
-import { AnalyticsServiceBase } from "../../common/services/analytics-service-base";
 import { ChildProcess } from "child_process";
 import * as path from "path";
 import { cache } from "../../common/decorators";
-import { isInteractive } from '../../common/helpers';
+import { isInteractive, toBoolean } from '../../common/helpers';
 import { DeviceTypes, AnalyticsClients } from "../../common/constants";
 
-export class AnalyticsService extends AnalyticsServiceBase {
+export class AnalyticsService implements IAnalyticsService, IDisposable {
 	private static ANALYTICS_BROKER_START_TIMEOUT = 10 * 1000;
 	private brokerProcess: ChildProcess;
-
-	constructor(protected $logger: ILogger,
-		protected $options: IOptions,
-		protected $processService: IProcessService,
-		$staticConfig: Config.IStaticConfig,
-		$prompter: IPrompter,
-		$userSettingsService: UserSettings.IUserSettingsService,
-		$analyticsSettingsService: IAnalyticsSettingsService,
-		$osInfo: IOsInfo,
+	private shouldDisposeInstance: boolean = true;
```

Comment:
not used?

Reply:
It's used [here](https://github.com/NativeScript/nativescript-cli/pull/4153/files#diff-d360c9e83fa92466da673c7b7366132dR134) - in the dispose method